### DISCUSSION TITLE
yarn install hanging

### DISCUSSION THREAD

**Discussion by SirUnreal** [mastodon/mastodon#24902] [General]
Update to version 4.1.2:

- updated ruby to 3.0.6
- checked out version 4.1.2
- bundle install successfully executed
- yarn install hangs for hours, process "ruby" has 100% cpu load
- but after restart of the Mastodon services the version is now on 4.1.2
- restarted even the server, but yarn install is not working

What does yarn install do? Is my instance successfully updated?

**Comment by renchap** [comment]
`yarn` is a Javascript package manager, it should not involve Ruby at all.

I am not sure what is happening here, but you might want to check your NodeJS and yarn install.

**Reply by SirUnreal** [reply]
Yes, I had problems with yarn after a clean install of Mastodon 4.0.x in 2022. Don't know the details anymore, but yarn could not be executed. So I installed it via Debian repo and softlinked it from ~mastodon/live/bin/yarn to /usr/bin/yarn. Now I see (thanks for the hint!) after the steps above that the softlink is gone. So I got back the strange yarn situation from the beginning. Under https://docs.joinmastodon.org/admin/install/ I don't see a solution. Should I "apt remove yarn"? How can I re-install it under the mastodon account?

**Reply by renchap** [reply]
Sorry for the delay, I missed your response!

Yes, you should remove Yarn from Debian, and if you are running a recent NodeJS version (16.20+ I think?) use Corepack (this is coming from Node) to install yarn, as advised in the documentation:

```
corepack enable
yarn set version classic
```

**Reply by SirUnreal** [reply]
Today for v4.1.3 yarn was removed via apt, but the problem is still there. With ps I can see that "ruby /home/mastodon/live/bin/yarn install" takes 100% cpu.

mastodon@xxxxxx:~/live$ ruby -rdebug /home/mastodon/live/bin/yarn install
/appdata/mastodon/.rbenv/versions/3.0.6/lib/ruby/3.0.0/debug.rb:6:in `<top (required)>': undefined method `>' for nil:NilClass (NoMethodError)
	from <internal:/appdata/mastodon/.rbenv/versions/3.0.6/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/appdata/mastodon/.rbenv/versions/3.0.6/lib/ruby/3.0.0/rubygems/core_ext/kernel_require.rb>:85:in `require'

I have a softlink at /home/mastodon linked to /appdata/mastodon - in case that is a problem (which should not). Reinstalled rbenv and bundler, nothing helped.

Now I reinstalled yarn via apt. Softlinked it from /home/mastodon/live/bin/yarn to /usr/bin/yarn and yarn install runs until the end.
